apps/action_data_fetcher/lib/action_data_fetcher/gpo/fetcher/worker.ex

```diff
@@ -0,0 +1,95 @@
+defmodule ActionDataFetcher.GPO.Fetcher.Worker do
+  use GenServer
+
+  @http Application.get_env(:action_data_fetcher, :gpo)[:http_client] || HTTPoison
+  @file_module Application.get_env(:action_data_fetcher, :gpo)[:file_module] || File
+  @sys_module Application.get_env(:action_data_fetcher, :gpo)[:sys_module] || System
+  @tmp_dir Application.get_env(:action_data_fetcher, :tmp_dir)
+
+  defmodule FetchState do
+    defstruct [:congress, :bill_type, :zip_data]
+  end
+
+  ## Worker API
+
+  def start_link(stuff) do
+    GenServer.start_link(__MODULE__, stuff)
+  end
+  
+  def fetch_bills(pid, congress, bill_type) do
+    GenServer.call(pid, {:fetch_bills, {:congress, congress, :bill_type, bill_type}})
+  end
+
+  def init(_state) do
+    {:ok, %FetchState{}}
+  end
+
+  def handle_call({:fetch_bills, {:congress, congress, :bill_type, bill_type}}, _from, _state) do
+    {:reply, fetch_bills(congress, bill_type), %FetchState{}}
+  end
+
+  ## Internal Helpers
+
+  defp fetch_bills(congress, bill_type) do
+	case fetch_zip(congress, bill_type) do
+      {:ok, %HTTPoison.Response{status_code: 200, body: zip}} ->
+        write_zip_to_tmpdir(congress, bill_type, zip)
+        |> unzip_bill_data
+        |> cleanup
+      {:ok, %HTTPoison.Response{status_code: 404}} -> {:error, :not_found}
+        _ -> {:error, :unknown}
+    end
+  end
+
+  defp fetch_zip(congress, type) do
+    generate_url(congress, type) 
+    |> @http.get([timeout: 90_000]) # TODO: Magic constant
+  end
+
+  defp generate_url(congress, bill_type) do
+    # TODO: should this also be configured?
+    "https://www.gpo.gov/fdsys/bulkdata/BILLSTATUS/#{congress}/#{bill_type}/BILLSTATUS-#{congress}-#{bill_type}.zip"
+  end
+
+  defp write_zip_to_tmpdir(congress, bill_type, zip_data) do
```

Comment:
it's not clear to me why the tmp file is necessary. Can we just use http://elixir-lang.org/getting-started/erlang-libraries.html#the-zip-and-zlib-modules to do this in memory? We're not going to rely on the file artifacts in any way, are we?